src/app/modules/main/wallet_section/filter.nim

```diff
@@ -27,17 +27,11 @@ proc `$`*(self: Filter): string =
 
 proc setFillterAllAddresses*(self: Filter) = 
   self.allAddresses = true
-  self.addresses = self.controller.getWalletAccounts().map(a => a.address)
-
-proc toggleWatchOnlyAccounts*(self: Filter) =
-  self.controller.toggleIncludeWatchOnlyAccount()
-
-proc includeWatchOnlyToggled*(self: Filter) =
-  let includeWatchOnly = self.controller.isIncludeWatchOnlyAccount()
-  if includeWatchOnly:
-    self.setFillterAllAddresses()
-  else:
-    self.addresses = self.controller.getWalletAccounts().filter(a => a.walletType != "watch").map(a => a.address)
+  var allAccounts = self.controller.getWalletAccounts()
+  var accountsExclWatchAccs = allAccounts.filter(a => not a.hideFromTotalBalance)
+  if allAccounts.len != accountsExclWatchAccs.len:
```

Comment:
@Khushboo-dev-cpp I am not sure I get the purpose of this condition, could you explain it?
As I see this condition always evaluates to `true`.

Reply:
so basically this condition check wether all accounts are being shown or there some of them that hidden because of  the hideFromTotalBalance property, is used to adapt show valid activity and collectibles.
this can be false in case all watched accounts are shown in total balance